src/components/Slot/Slot.tsx

```diff
@@ -12,12 +12,18 @@ export interface ISlotProps {
   variables?: ComponentVariablesInput
 }
 
+export const createSlotFactory = (as: any, mapValueToProps: MapValueToProps) => (
+  val,
+  options: ObjectOf<any> = {},
```

Comment:
this type changed a bit, pls take a look again at #326, and also apply the fix from `factories.tsx` cause you will have a weird type error related to `IProps`